packages/cli/src/aws/update-server.ts

```diff
@@ -1,32 +1,52 @@
-import { UpdateServiceCommand } from '@aws-sdk/client-ecs';
-import { ecsClient, getEcsServiceName, getStackByTag } from './utils';
+import { getServerVersions } from './utils';
+import { configFileName, readConfig, writeConfig } from '../utils';
+import { MedplumInfraConfig } from '@medplum/core';
+import * as semver from 'semver';
+import { spawnSync } from 'child_process';
+import { createMedplumClient } from '../util/client';
 
 /**
  * The AWS "update-server" command updates the Medplum server in a Medplum CloudFormation stack.
  * @param tag - The Medplum stack tag.
+ * @param options - Client options
  */
-export async function updateServerCommand(tag: string): Promise<void> {
-  const details = await getStackByTag(tag);
-  if (!details) {
-    console.log('Stack not found');
+export async function updateServerCommand(tag: string, options: any): Promise<void> {
+  const client = await createMedplumClient(options);
+  const config = readConfig(tag) as MedplumInfraConfig;
+  if (!config) {
+    console.log(`Configuration file ${configFileName(tag)} not found`);
     return;
   }
-  const ecsCluster = details.ecsCluster?.PhysicalResourceId;
-  if (!ecsCluster) {
-    console.log('ECS Cluster not found');
-    return;
+  const separatorIndex = config.serverImage.lastIndexOf(':');
+  const initialVersion = config.serverImage.slice(separatorIndex + 1);
+  let updateVersion = await nextUpdateVersion(initialVersion);
+  while (updateVersion) {
+    console.log(`Performing update to v${updateVersion}`);
+    config.serverImage = `${config.serverImage.slice(0, separatorIndex)}:${updateVersion}`;
+    deployServerUpdate(tag, config);
+    // Run data migrations
+    await client.post('/admin/super/migrate', '');
+
+    updateVersion = await nextUpdateVersion(updateVersion);
   }
-  const ecsService = getEcsServiceName(details.ecsService);
-  if (!ecsService) {
-    console.log('ECS Service not found');
-    return;
+}
+
+async function nextUpdateVersion(currentVersion: string): Promise<string | undefined> {
+  return (await getServerVersions(currentVersion))
+    .filter((v) => semver.gte(v, semver.inc(currentVersion, 'minor') as string))
+    .pop();
+}
+
+function deployServerUpdate(tag: string, config: MedplumInfraConfig): void {
+  const configFile = configFileName(tag);
+  writeConfig(config, tag);
+
+  const cmd = `npx cdk deploy -c config=${configFile}${config.region !== 'us-east-1' ? ' --all' : ''}`;
+  console.log('> ' + cmd);
+  const deploy = spawnSync(cmd);
```

Comment:
Does `spawnSync()` allow interactivity with the spawned process?  I.e., if `cdk` prompts the user "Is this OK?", will you still be able to press "y" to proceed?

Reply:
Actually you might wanna check out `inherit`, the default behavior will not keep the binding to the current `pty`

Reply:
```suggestion
  const deploy = spawnSync(cmd, { stdio: 'inherit' });
```

Reply:
Thanks @ThatOneBro — I misread the Node docs 😅